src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -696,19 +713,18 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
     // - Helper function to parse the preferred shell type from user settings returned by cloud console API.
     // We need this function because the field might be missing in the settings
     // created with old versions of cloud console API.
-    std::optional<utility::string_t> AzureConnection::_ParsePreferredShellType(const web::json::value& settingsResponse)
+    winrt::hstring AzureConnection::_ParsePreferredShellType(const WDJ::JsonObject& settingsResponse)
     {
-        if (settingsResponse.has_object_field(L"properties"))
+        if (settingsResponse.HasKey(L"properties"))
         {
-            const auto userSettings = settingsResponse.at(L"properties");
-            if (userSettings.has_string_field(L"preferredShellType"))
+            const auto userSettings = settingsResponse.GetNamedObject(L"properties");
+            if (userSettings.HasKey(L"preferredShellType"))
             {
-                const auto preferredShellTypeValue = userSettings.at(L"preferredShellType");
-                return preferredShellTypeValue.as_string();
+                return userSettings.GetNamedString(L"preferredShellType");
             }
         }
 
-        return std::nullopt;
+        return winrt::hstring{};

```

Comment:
Wouldn't it make sense to just return "pwsh" here?

Reply:
Done&done, good idea